technology/sql/src/main/java/com/github/mizool/technology/sql/jooq/ResultQueries.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 incub8 Software Labs GmbH
+ * Copyright 2020 protel Hotelsoftware GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mizool.technology.sql.jooq;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import lombok.experimental.UtilityClass;
+
+import org.jooq.ResultQuery;
+import org.jooq.Table;
+import org.jooq.TableRecord;
+import org.jooq.exception.DataAccessException;
+
+import com.github.mizool.core.exception.StoreLayerException;
+
+@UtilityClass
+public class ResultQueries
+{
+    public <R extends TableRecord<R>> Optional<R> fetchOptional(ResultQuery<R> resultQuery, Table<R> table)
```

Comment:
`fetchOptional()` is named like the original JOOQ method, but I find the name to be misleading: it makes these two methods look like you can choose between "an optional" and "a stream" when in fact you are fetching either **a single record** (our classic `read` store operation, and I imagine it would also be used for `findBy...`) or **multiple records** (our classic `list` store operations).

I suggest we rename this to `fetchOptional()` to `read()` and renaming `fetchStream()` to `list()`. Alternatively, one could use something like `fetchSingle()` and `fetchMultiple()`, but I prefer the read/list wording.